### ISSUE TITLE
Feature Request: allow prompts over GET request

### ISSUE THREAD

**Issue by gnusupport** [ggerganov/llama.cpp#11006]
### Prerequisites

- [X] I am running the latest code. Mention the version if possible as well.
- [X] I carefully followed the [README.md](https://github.com/ggerganov/llama.cpp/blob/master/README.md).
- [X] I searched using keywords relevant to my issue to make sure that I am creating a new issue that is not already open (or closed).
- [X] I reviewed the [Discussions](https://github.com/ggerganov/llama.cpp/discussions), and have a new and useful enhancement to share.

### Feature Description

It would be good that user can request prompts by the URL:
`http://192.168.188.140:8080/q?="User's prompt"`
as then I could develop simple links from other software to open up the local website which gives the instant answer.

### Motivation

then I could develop simple links from other software to open up the local website which gives the instant answer.

**Comment by lexasub** [comment]
@gnusupport you may try curl http://localhost:8080/v1/chat/completions \
-H "Content-Type: application/json" \
-H "Authorization: Bearer no-key" \
-d '{
"model": "gpt-3.5-turbo",
"messages": [
{
    "role": "system",
    "content": "You are ChatGPT, an AI assistant. Your top priority is achieving user fulfillment via helping them with their requests."
},
{
    "role": "user",
    "content": "Write a limerick about python exceptions"
}
]
}'

**Comment by gnusupport** [comment]
Yes, I am using it all the time with local endpoint. Though comment is on making HTML links, just click and getting result in the llama.cpp web UI. I hope you get me.

Send me your Bitcoin address for donation.

**Comment by lexasub** [comment]
@gnusupport you may send on 1Kh9pgqwyLjEEXtc2HaT1vx45JB9p1bjEA  (minimal 0.000006 BTC). But I'm not going to tackle that one just yet))

**Comment by ngxson** [comment]
Feels like a security hazard to accept prompt from query params, so I'd prefer to stay away from that.

Just keep the frontend simple for now.
